node/coinstacks/ethereum/api/src/controller.ts

```diff
@@ -147,29 +147,58 @@ export class Ethereum extends Controller implements BaseAPI, EthereumAPI {
   @Get('account/{pubkey}/txs')
   async getTxHistory(
     @Path() pubkey: string,
-    @Query() page?: number,
-    @Query() pageSize = 25,
-    @Query() contract?: string
-  ): Promise<TxHistory> {
+    @Query() cursor?: string,
+    @Query() pageSize = 25
+  ): Promise<EthereumTxHistory> {
     try {
-      const data = await blockbook.getAddress(pubkey, page, pageSize, undefined, undefined, 'txs', contract)
+      let page = 1
+      if (cursor) {
+        const cursorDecoded = Buffer.from(cursor, 'base64').toString('binary')
+        if (isNaN(Number(cursorDecoded))) {
+          throw new ApiError('Bad Request', 400, 'Invalid cursor, please use the cursor returned in previous page')
+        }
+        page = Number(cursorDecoded)
+      }
+
+      const data = await blockbook.getAddress(pubkey, page, pageSize, undefined, undefined, 'txs')
+
+      const nextPage = page + 1
+
+      const nextCursor =
```

Comment:
We can utilize the totalPages response from blockbook to determine if there are more pages instead of using the length of transactions returned.

Theres a possible edgecase in which the final page returns a full page size which will then be treated as if there is another page even though it was the final transaction.